.github/workflows/debian_10_arm.yml

```diff
@@ -0,0 +1,208 @@
+
+      - uses: ./.github/actions/handle_docker
+        id: handle_docker
+        with:
+          images: |
+            ov_build/debian_10
+          registry: 'openvinogithubactions.azurecr.io'
+          dockerfiles_root_dir: '.github/dockerfiles'
+          changed_components: ${{ needs.smart_ci.outputs.changed_components }}
+
+  Build:
+    needs: [ Smart_CI, Docker ]
+    timeout-minutes: 150
+    defaults:
+      run:
+        shell: bash
+    runs-on: 'aks-linux-16-cores-arm'
+    container:
+      image: ${{ fromJSON(needs.docker.outputs.images).ov_build.debian_10 }}
+      volumes:
+        - /mount:/mount
+      options: -e SCCACHE_AZURE_BLOB_CONTAINER -e SCCACHE_AZURE_CONNECTION_STRING
+    env:
+      CMAKE_BUILD_TYPE: 'Release'
+      CMAKE_CXX_COMPILER_LAUNCHER: sccache
+      CMAKE_C_COMPILER_LAUNCHER: sccache
+      SCCACHE_IGNORE_SERVER_IO_ERROR: 1
+      SCCACHE_SERVER_PORT: 35555
+      SCCACHE_ERROR_LOG: /__w/openvino/sccache_log.txt
+      SCCACHE_LOG: warn
+      GITHUB_WORKSPACE: '/__w/openvino/openvino'
+      OPENVINO_REPO: /__w/openvino/openvino/openvino
+      INSTALL_DIR: /__w/openvino/openvino/openvino_install
+      INSTALL_TEST_DIR: /__w/openvino/openvino/tests_install
+      BUILD_DIR: /__w/openvino/openvino/openvino_build
+      SCCACHE_AZURE_KEY_PREFIX: 'debian_10_arm_Release'
+    if: "!needs.smart_ci.outputs.skip_workflow"
+
+    steps:
+      - name: Clone OpenVINO
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          path: ${{ env.OPENVINO_REPO }}
+          submodules: 'true'
+
+      #
+      # Print system info
+      #
+
+      - name: System info
+        uses: ./openvino/.github/actions/system_info
+
+      #
+      # Dependencies
+      #
+
+      - name: Install python dependencies
+        run: |
+          python3 -m pip install -r ${OPENVINO_REPO}/tools/mo/requirements.txt
+          python3 -m pip install -r ${OPENVINO_REPO}/src/bindings/python/requirements_test.txt
+          python3 -m pip install -r ${OPENVINO_REPO}/src/bindings/python/wheel/requirements-dev.txt
+
+      #
+      # Build
+      #
+
+      - name: CMake configure - OpenVINO
+        run: |
+          cmake \
+            -DCMAKE_TOOLCHAIN_FILE=${OPENVINO_REPO}/cmake/arm.toolchain.cmake \
+            -DTHREADS_PTHREAD_ARG="-pthread" \
+            -DCMAKE_COMPILE_WARNING_AS_ERROR_DEFAULT=OFF \
+            -DCMAKE_BUILD_TYPE=${{ env.CMAKE_BUILD_TYPE }} \
+            -DENABLE_STRICT_DEPENDENCIES=OFF \
+            -DCMAKE_VERBOSE_MAKEFILE=ON \
+            -DENABLE_CONFORMANCE_PGQL=ON \
+            -DENABLE_LTO=ON \
+            -DENABLE_TESTS=ON \
+            -DENABLE_PYTHON=OFF \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=${{ env.CMAKE_CXX_COMPILER_LAUNCHER }} \
+            -DCMAKE_C_COMPILER_LAUNCHER=${{ env.CMAKE_C_COMPILER_LAUNCHER }} \
+            -S ${OPENVINO_REPO} \
+            -B ${BUILD_DIR}
```

Comment:
do you think we can reuse reusable workflow for Linux build? 
all Linux builds should loo the same

I would try existing Linux workflow as is.

Reply:
I looked into it:
* The `job_build_linux.yml` is bigger and thus longer with stages like building OV Contrib, JS API, etc. - are they ok to be executed with Debian? If not, we might skip them via `os` input.
* The Debian ARM utilizes the toolchain file `${OPENVINO_REPO}/cmake/arm.toolchain.cmake`. It might be selected based on the same `os` input. Is there any toolchain file we could use for other Linux builds so that the rest of workflows could also benefit from the `os == 'debian' && toolchain=".../one" || toolchain=".../another"`

Reply:
what if we add parameters to build job?
- cmake arguments. This will allow to specify cmake toolchain and some other arguments
- whether to build contrb
- whether to build JS
- whether to build Debian packages